CatalogAPI/Extensions/PaginationExtension.cs

```diff
@@ -0,0 +1,63 @@
+﻿using CatalogAPI.CommonModel;
+using CatalogAPI.Configuration;
+namespace CatalogAPI.Extensions
+{
+    public static class PaginationExtension
+    {
+        public static Pagination<T> ToPagedList<T>(this IEnumerable<T> data, int page, int pageSize)
```

Comment:
`IEnumerable` implies enumerating a collection once. If you need, count, first, last pass an `ICollection`

Reply:
Changed with ICollection.